cirq/ops/parallel_gate_operation.py

```diff
@@ -0,0 +1,233 @@
+                # funky behavior of fraction, cast to str in constructor helps.
+                approx_frac = Fraction(info.exponent).limit_denominator(16)
+                if approx_frac.denominator not in [2, 4, 5, 10]:
+                    if abs(float(approx_frac)
+                           - info.exponent) < 10 ** -args.precision:
+                        return '({})'.format(approx_frac)
+
+                return '{{:.{}}}'.format(args.precision).format(info.exponent)
+        return repr(info.exponent)
+
+    def _circuit_diagram_info_(self,
+                               args: protocols.CircuitDiagramInfoArgs
+                               ) -> protocols.CircuitDiagramInfo:
+        diagram_info = protocols.circuit_diagram_info(self.gate,
+                                                      args,
+                                                      NotImplemented)
+        if diagram_info == NotImplemented:
+            return diagram_info
+
+        # Extract the formatted exponent and place it in wire symbols so that
+        # every gate on the diagram has an exponent instead of the last one
+        symbol = diagram_info.wire_symbols[0]
+        exponent = self._formatted_exponent(diagram_info, args)
+        if exponent is not None:
+            symbol = "{0}^{1}".format(symbol, exponent)
+
+        # Include every gate on the diagram instead of one
+        wire_symbols = (symbol,) * len(self.qubits)
+        # Set exponent=1 so that another extra exponent doesn't get tacked on
+        return protocols.CircuitDiagramInfo(wire_symbols=wire_symbols,
+                                            exponent=1,
+                                            connected=False)
+
+    def _phase_by_(self, phase_turns: float,
+                   qubit_index: int) -> 'ParallelGateOperation':
+        phased_gate = protocols.phase_by(self._gate, phase_turns, qubit_index,
+                                         default=None)
+        if phased_gate is None:
+            return NotImplemented
+        return self.with_gate(phased_gate)
+
+    def _trace_distance_bound_(self) -> float:
+        return 1.
+
+    def __pow__(self, exponent: Any) -> 'ParallelGateOperation':
+        """Raise gate to a power, then reapply to the same qubits.
+
+        Only works if the gate implements cirq.ExtrapolatableEffect.
+        For extrapolatable gate G this means the following two are equivalent:
+
+            (G ** 1.5)(qubit)  or  G(qubit) ** 1.5
+
+        Args:
+            exponent: The amount to scale the gate's effect by.
+
+        Returns:
+            A new operation on the same qubits with the scaled gate.
+        """
+        new_gate = protocols.pow(self.gate,
+                                 exponent,
+                                 NotImplemented)
+        if new_gate is NotImplemented:
+            return NotImplemented
+        return self.with_gate(new_gate)
+
+    def _qasm_(self, args: protocols.QasmArgs) -> Optional[str]:
```

Comment:
I believe this will work better if you don't implement `_qasm_` at all, because the qasm output code will auto decompose the gate and helpfully comment that it came from a parallel operation. Test it out.

Reply:
Removing the does provide qasm code with a neat comment specifying the the code block below it corresponds to the same operation. I like the aesthetics of that more, so I'll remove this implementation.